app/src/main/java/com/example/byitscover/scrapers/BarnesAndNobleScraper.java

```diff
@@ -43,50 +42,25 @@ public BarnesAndNobleScraper() {
      * @throws IOException
      */
     public List<BookListing> scrape(Query query) throws IOException {
-        List<Result> results = ScraperHelper.googleAPISearch(ScraperConstants.BARNES_AND_NOBLE, query);
-        String searchingUrl = getActualBookResult(results);
-        Document document = Jsoup.connect(searchingUrl).get();
-        System.out.println(searchingUrl);
+        String url = getUrlWithQuery(query);
+        System.out.println(url);
+        Document document = Jsoup.connect(url).get();
+        Elements bookLinks = document.getElementsByClass("pImageLink ");
 
-        Map<String, String> toReturn = new HashMap<String, String>();
+        List<BookListing> listings = new ArrayList<BookListing>();
 
-        //get rating by parsing json js var
-        String htmlString = document.html();
-        Double rating = null;
-        String reviewValueString;
+        BookListing firstListing = getListingFromElement(bookLinks, 0, query);
+        listings.add(firstListing);
 
-        try {
-            rating = Double.parseDouble(htmlString.substring(htmlString.indexOf(",\"rating\":") + 10,
-                    htmlString.indexOf(",\"rating\":") + 13));
-        } catch (Exception ex) {
-            rating = 0.0;
+        if (listings.size() > 1) {
+            BookListing secondListing = getListingFromElement(bookLinks, 1, query);
+            listings.add(secondListing);
         }
-
-
-        try {
-            //get review
-            Element reviewValue = (Element) document.getElementById("overviewSection").childNode(1).childNode(1)
-                    .childNode(3).childNode(1).childNode(1).childNode(1);
-            reviewValueString = Jsoup.clean(reviewValue.toString(), Whitelist.none());
-        } catch (Exception ex) {
-            reviewValueString = "";
+        if (listings.size() > 1) {
```

Comment:
This should be `listings.size() > 2`

Reply:
I believe it should be >1 since it's the second listing